Softeq.XToolkit.WhiteLabel.Droid/ViewComponents/ViewModelComponent.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using AndroidX.Fragment.App;
+
+namespace Softeq.XToolkit.WhiteLabel.Droid.ViewComponents
+{
+    /// <summary>
+    ///     Component, which encapsulates logic of storing/restoring ViewModel instance within FragmentManager
+    /// </summary>
+    /// <typeparam name="TViewModel">
+    ///     Type of the ViewModel instance
+    /// </typeparam>
+    public sealed class ViewModelComponent<TViewModel> where TViewModel : class
```

Comment:
It can be better to rename this class, like `ViewModelStoreComponent` or something else

Reply:
The idea behind this class is not only to store ViewModel, but completely control ViewModel's lifecycle within Fragment. The main purpose is to provide ViewModel to a Fragment whenever possible, Save/Restore/Clear methods are just callbacks to the Fragment's lifecycle, basically.

Reply:
`ViewModelComponent` is too general
what about `ViewModelLifecycleComponent`/`ViewModelFragmentLifecycleComponent`?

Reply:
Again, controlling lifecycle is just a way to achieve main goal, and the main goal of the component - is to safely provide ViewModel instance whenever possible. It is basically a wrapper around ViewModel property in Fragments. It is what Fragments (well, at least DialogFragments) should use instead of storing reference to ViewModel directly. So it is a component, which provides ViewModel instance to whoever uses it.